pkg/ingester/pyroscope/ingest_adapter.go

```diff
@@ -152,6 +159,55 @@ func (p *pyroscopeIngesterAdapter) Evaluate(input *storage.PutInput) (storage.Sa
 	return nil, false // noop
 }
 
+func (p *pyroscopeIngesterAdapter) parseToPprof(ctx context.Context, in *ingestion.IngestInput, pprofable ingestion.ParseableToPprof) error {
+	plainReq, err := pprofable.ParseToPprof(ctx, in.Metadata)
```

Comment:
Anything in that err should translate into 4xx eg. if JFR can't be parsed...

Reply:
This is exactly like that. I will add a unit test to double check

Reply:
https://github.com/grafana/pyroscope/pull/2355/commits/5970622861a163d6cec54677feec6fe3b7482e64